ATOMIC_SWAP.md

```diff
@@ -0,0 +1,259 @@
+        http://localhost:8000/wallet/atomic-swap/tx-proposal/sign-and-push
+{
+  "success": true,
+  "hash": "00004fb4e4b2db216e6314f1e6e9be73118b3974ea7180044edf5851d0a31045",
+  "inputs": [
+    {
+      "hash": "00c7797738d890d1517f637f2af079d8daa4449fdfd098d2d636bb0e30c809d1",
+      "index": 1,
+      "data": {
+        "type": "Buffer",
+        "data":[...]
+      },
+      "tx_id": "00c7797738d890d1517f637f2af079d8daa4449fdfd098d2d636bb0e30c809d1"
+    }
+  ],
+  "outputs": [
+    {
+      "value": 1,
+      "script": {
+        "type": "Buffer",
+        "data": []
+      },
+      "tokenData": 0,
+      "decodedScript": {
+        "address": {
+          "base58": "WXf4xPLBn7HUC7F1U2vY4J5zwpsDS12bT6",
+          "network": {...}
+        },
+        "timelock": null
+      },
+      "token_data": 0
+    }
+  ],
+  "version": 1,
+  "weight": 17.13552634501032,
+  "nonce": 95503,
+  "timestamp": 1647035043,
+  "parents": [
+    "007ae58167abb0e63d6879c437659c7f20ecc8a74531bcc612367735bbc191cc",
+    "000489804c1924510d3a2fb5e9a5b71df6e82436dc74fa4b7bfae72bd5502805"
+  ],
+  "tokens": [],
+}
+```
+
+## Multisig
+
+If a MultiSig wallet wants to participate in an atomic-swap the signing process is a bit different since the signature of a MultiSig requires a set number of participants signatures.
+
+Once you arrive at the signing phase you should extract the transaction hex from the proposal string.
+
+The transaction hex is the second element when spliting the proposal string by the `|` character.
+Proposal string: `PartialTx|<txHex>|...|...`
+
+## Lock mechanism
+
+When creating and updating proposals the default behavior is to lock the inputs added (i.e. mark as already selected).
+The lock mechanism is a safety measure so that the proposal stays valid, even if the wallet is used for sending tokens or if there are multiple on-going atomic-swaps.
+This lock does not persist after a restart and if an input of the proposal is spent for any reason, the previously created proposal will be voided.
+
+But for development purposes or if the caller wants to make a "simulation" (i.e. check if there is no issues when creating the transaction like lack of funds) the lock mechanism can be bypassed.
+The way to bypass is to send the parameter `lock` as `false` when creating/updating the tx-proposal.
+This `lock=false` will only be valid for the operations on the call it is made.
+
+If, for any reason, the user does not want to keep the inputs locked for this proposal they can unlock the inputs with the `/wallet/atomic-swap/tx-proposal/unlock` API.
+
+### POST /wallet/atomic-swap/tx-proposal/unlock
+
+Will unlock all inputs of the caller's wallet present on the proposal string.
+
+```bash
+$ curl  -X POST \
+        -H "X-Wallet-Id: 123" \
+        -H "Content-type: application/json" \
+        --data "{\"partial_tx\":\"<proposalString>\"}" \
+        http://localhost:8000/wallet/atomic-swap/tx-proposal/unlock
+{
+  "success": true,
+}
+```
```

Comment:
I like the description above but I feel we should add a new section with a more detailed and verbose example.

Bob and Alice would like to exchange Token A (TKNA) and HTR. They agree that Bob will send 10 HTR and receive 2 TKNA. Bob will then start the atomic swap creating the tx-proposal, (tx proposal code to send 10 HTR and receive 2 TKNA), get the serialized data, sends to Alice, Alice decodes, validates it's correct, adds it's own output and fills the inputs, checks tx is completed and correct, get her signatures, send back to Bob the full tx with her signatures. Bob decodes the tx, validates, add his own signatures, push to the network.

But with verbose details on each step. I would also add a second example with one of the atomic swap wallets being a multisig one, I think we need this in more details rather than just saying it's possible and briefly mentioning how to do it.